docs/source/firstDraft.py

```diff
@@ -1,17 +1,56 @@
-from qgis.core import QgsProject
-
-csv = QgsProject.instance().mapLayersByName('SC_D_POZEMKU')[0]
-csvField = 'KOD'
-layer = QgsProject.instance().mapLayersByName('Parcely')[0]
-layerField = 'DruhPozemkuKod'
-
-joinObject = QgsVectorLayerJoinInfo()
-joinObject.setJoinLayerId(csv.id())
-joinObject.setJoinFieldName(csvField)
-joinObject.setTargetFieldName(layerField)
-joinObject.setPrefix('g')
-joinObject.setUsingMemoryCache(True)
-joinObject.setJoinLayer(csv)
-
-layer.addJoin(joinObject)
-QgsProject.instance().addMapLayer(layer)
\ No newline at end of file
+import os
+from qgis.core import QgsProject, QgsVectorLayer, QgsVectorLayerJoinInfo
+
+def add_join_csv_file(target_layer: QgsVectorLayer, csv_layer: QgsVectorLayer, target_field: str = "DruhPozemkuKod", csv_field: str = "KOD", csv_subset: list = ["NAZEV"]) -> QgsVectorLayer:
```

Comment:
U parametru  `csv_layer` by bylo praktictejsi, kdyby slo o nazev ciselniku, napr 

```
 mylayer = add_join_csv_file(parcely_layer, "druh_pozemku", ...)
```

Objekt QgsVectorLayer by vytvorila lokalne primo funkce `add_join_csv_file`.

Reply:
Osobně mně to nepřijde praktičtější: v případě předávání jen názvu číselníku musí funkce `add_join_csv_file` ošetřovat mnohem víc parametrů, jako umístění číselníku nebo oddělovač. Dává mně větší smysl přejmenovat funkci na `add_join`, jí předávat dva objekty QgsVectorLayer, u nichž správnost zajišťuje uživatel a současnou funkcionalitu ponechat.